lib/ansible/modules/system/xfs_quota.py

```diff
@@ -0,0 +1,367 @@
+      - the name of the user, group or project to apply the quota to, if other than default
+  mountpoint:
+    description:
+      - the mountpoint on which to apply the quotas
+    required: true
+  bhard:
+    description:
+      - hard blocks quota limit
+  bsoft:
+    description:
+      - soft blocks quota limit
+  ihard:
+    description:
+      - hard inodes quota limit
+  isoft:
+    description:
+      - soft inodes quota limit
+  rtbhard:
+    description:
+      - hard realtime blocks quota limit
+  rtbsoft:
+    description:
+      - soft realtime blocks quota limit
+  state:
+    description:
+      - Whether to apply the limits or remove them.
+      - When removing limit, they are set to 0, and not quite removed.
+    default: present
+    choices:
+      - present
+      - absent
+
+notes:
+   - Not tested on any debian based system
+requirements:
+   - xfsprogs
+'''
+
+EXAMPLES = r'''
+- name: Set default project soft and hard limit on /opt of 1g
+  xfs_quota:
+    type: project
+    mountpoint: /opt
+    bsoft: 1g
+    bhard: 1g
+    state: present
+
+- name: Remove the default limits on /opt
+  xfs_quota:
+    type: project
+    mountpoint: /opt
+    state: absent
+
+- name: Set default soft user inode limits on /home of 1024 inodes and hard of 2048
+  xfs_quota:
+    type: user
+    mountpoint: /home
+    isoft: 1024
+    ihard: 2048
+
+'''
+
+RETURN = ''' # '''
+
+import os
+import sys
+import json
+import pwd
+import grp
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import human_to_bytes
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            type=dict(required=True, choices=['user', 'group', 'project']),
+            name=dict(required=False, default=None),
+            mountpoint=dict(required=True),
+            bhard=dict(required=False, default=None),
+            bsoft=dict(required=False, default=None),
+            ihard=dict(required=False, default=None),
+            isoft=dict(required=False, default=None),
+            rtbhard=dict(required=False, default=None),
+            rtbsoft=dict(required=False, default=None),
+            state=dict(required=False, default='present', choices=['present', 'absent'])
+        ),
+        supports_check_mode=True
+    )
+
+    quota_type = module.params['type']
+    name = module.params['name']
+    mountpoint = module.params['mountpoint']
+    bhard = module.params['bhard']
+    bsoft = module.params['bsoft']
+    ihard = module.params['ihard']
+    isoft = module.params['isoft']
+    rtbhard = module.params['rtbhard']
+    rtbsoft = module.params['rtbsoft']
+    state = module.params['state']
+
+    if bhard is not None:
+        bhard = human_to_bytes(bhard)
+
+    if bsoft is not None:
+        bsoft = human_to_bytes(bsoft)
+
+    if rtbhard is not None:
+        rtbhard = human_to_bytes(rtbhard)
+
+    if rtbsoft is not None:
+        rtbsoft = human_to_bytes(rtbsoft)
+
+    changed = False
```

Comment:
It is better to have a **result** dictionary with all the information you want to return.
```suggestion
    result = dict(
        changed=False,
    )
```

Reply:
done